### PR TITLE
Const tuple unpacking

### PR DESCRIPTION
**Pull request by nc-x**
Fixes https://github.com/nim-lang/Nim/issues/6355

While this works for all the cases I tried, there may still be bugs.
Review carefully.

### CODE REVIEW

File: compiler/ast.nim [hunk 1]
```diff
@@ -223,6 +223,7 @@ type
     nkBreakState,         # special break statement for easier code generation
     nkFuncDef,            # a func
     nkTupleConstr         # a tuple constructor
+    nkConstTuple,         # a ``const (a, b) = expr`` construct
```
Comment (Araq) [inline compiler/ast.nim]: ``macros.nim`` needs to be patched too.

File: compiler/parser.nim [hunk 2]
```diff
@@ -1764,21 +1764,43 @@ proc parseSection(p: var TParser, kind: TNodeKind,
...
+
+  addSon(result, p.emptyNode)
+  eat(p, tkParRi)
   eat(p, tkEquals)
   optInd(p, result)
   addSon(result, parseExpr(p))
   indAndComment(p, result)
 
+proc parseConstant(p: var TParser): PNode =
+  #| constant = identWithPragma (colon typeDesc)? '=' optInd expr indAndComment
```
Comment (Araq) [inline compiler/parser.nim]: This comment describes the grammar rule and stopped reflecting reality. ;-)

File: compiler/ast.nim (hunk 1 above)
Comment (timotheecour) [reply compiler/ast.nim]: maybe we should add a test that makes sure these stay in sync
(doesnt' have to be in this PR though)

File: compiler/parser.nim (hunk 2 above)
Comment (nc-x) [reply compiler/parser.nim]: Also the grammar needs to be written for constTuple

But since I have no idea about EBNF, I leave this to you 😄

**Pull request merged**